### ISSUE TITLE
🐛 Panic at paint.rs

### ISSUE THREAD

**Issue by xudyang1** [dandavison/delta#1839]
- [x] Please include the raw text output from git, so that we can reproduce the problem.

```diff
--- a/lua/config/options.lua
+++ b/lua/config/options.lua
@@ -2,6 +2,11 @@ local opt = vim.opt

 opt.spelllang = "en_us"

+opt.title = true
+opt.titlestring =
+  [[%t%( %m%)%( (%<%{execute('lua local ft=vim.bo.filetype print((ft=="help" or ft=="man") and ft or vim.fn.expand("%:~:.:h"))')})%)]]
+opt.titlelen = 27
+
 -- TODO: more on rg and fdfind and wildcards...
 opt.grepprg = [[rg --vimgrep --hidden --glob "!{**/.git/**}"]]
 opt.wildignore = {
```

- [x] A screenshot of Delta's output is often helpful also.
![image](https://github.com/user-attachments/assets/1b52158d-88fe-4052-b4d4-f5a03b067049)

**Comment by th1000s** [comment]
Thank you for the report!

Setting my terminal to a width of 137 (see `echo $COLUMNS`, so it wraps after `filetype pr↵`, also setting `↵` to the empty string), and running

    cat crash_diff | HOME=/ delta --no-gitconfig --side-by-side  --wrap-left-symbol=' '

doesn't reproduce it here. Using delta 0.18.1 on Linux. Can you try that command, and add other flags from your gitconfig if needed?

**Comment by bendem** [comment]
I have a panic that looks like the same problem using latest delta:

```
thread 'main' panicked at /home/demarteaub/.cargo/registry/src/index.crates.io-6f17d22bba15001f/git-delta-0.18.1/src/paint.rs:957:17:
String mismatch encountered while superimposing style sections: ':' vs '↵'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

With 92 columns and the following git config:

```ini
[pager]
    diff = delta

[delta]
    side-by-side = true
    inspect-raw-lines = true
    features = navigate
    hunk-header-style = file line-number syntax
```

Part of the diff to reproduce:

```diff
diff --git a/ansible/ansible_collections/vdl/devops/plugins/lookup/password.py b/ansible/ansible_collections/vdl/devops/plugins/lookup/password.py
index 358f9328..5a8a1dc2 100644
--- a/ansible/ansible_collections/vdl/devops/plugins/lookup/password.py
+++ b/ansible/ansible_collections/vdl/devops/plugins/lookup/password.py
@@ -83,6 +83,7 @@ class Secret(typing.TypedDict):
 
 
 class LookupModule(LookupBase):
+
     def run(self, terms, variables: HostVars, **kwargs) -> typing.List[Secret]:
         # First of all populate options,
         # this will already take into account env vars and ini config
@@ -126,6 +127,8 @@ class LookupModule(LookupBase):
             force_renew = variables.get("vdl.devops.password.force_renew")
             secret = None
             skip_existing = False
+            # 0 means vault must reject the secret create if a secret already exists
+            current_version = 0
 
             if force_renew == "*all" or force_renew == secret_path:
                 display.display(f"forcing renew of {secret_path}", color="purple")
```


The delta output:
```console
$ < test.diff delta
thread 'main' panicked at /home/demarteaub/.cargo/registry/src/index.crates.io-6f17d22bba15001f/git-delta-0.18.1/src/paint.rs:957:17:
String mismatch encountered while superimposing style sections: ':' vs '↵'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Δ ansible/ansible_collections/vdl/devops/plugins/lookup/password.py
────────────────────────────────────────────────────────────────────────────────────────────

────────────────────────────────────────────────────────────────────────────────────────────────────────┐
• ansible/ansible_collections/vdl/devops/plugins/lookup/password.py:83: class Secret(typing.TypedDict): │
────────────────────────────────────────────────────────────────────────────────────────────────────────┘
│ 83 │                                        │ 83 │
│ 84 │                                        │ 84 │
│ 85 │class LookupModule(LookupBase):         │ 85 │class LookupModule(LookupBase):

```
![image](https://github.com/user-attachments/assets/0129598f-6c2c-4b75-917f-1b5f52efc06b)

**Comment by xudyang1** [comment]
> Thank you for the report!
>
> Setting my terminal to a width of 137 (see `echo $COLUMNS`, so it wraps after `filetype pr↵`, also setting `↵` to the empty string), and running
>
> ```
> cat crash_diff | HOME=/ delta --no-gitconfig --side-by-side  --wrap-left-symbol=' '
> ```
>
> doesn't reproduce it here. Using delta 0.18.1 on Linux. Can you try that command, and add other flags from your gitconfig if needed?

I am using windows terminal with WSL2 (Ubuntu 22.04) in full screen mode. The maximum `$COLUMNS` is 136:

![image](https://github.com/user-attachments/assets/0c9cd299-aa96-473f-bd6a-01640f06acec)

Even when I set `export COLUMNS=137`, the diff still reports error.

Does delta only work for columns greater than 136? @th1000s

**Comment by xudyang1** [comment]
By running `cargo install git-delta --locked`, the issue is fixed with delta 0.18.2.
